lsst/ts/laser/csc.py

```diff
@@ -79,15 +104,20 @@ def assert_propagating(self, action):
 
         Parameters
         ----------
-        action
-            The command being sent.
+        action: str
+            The name of the command being sent.
+
+        Raises
+        ------
+        ExpectedError
 
         Returns
         -------
+        None
 
         """
-        if self.detailed_state != LaserDetailedStateEnum.PROPAGATINGSTATE:
-            raise salobj.ExpectedError(f"{action} not allowed in state {self.detailed_state}")
+        if self.detailed_state != LaserDetailedState.PROPAGATINGSTATE:
+            raise ExpectedError(f"{action} not allowed in state {self.detailed_state}")
 
     async def do_changeWavelength(self,id_data):
```

Comment:
This method is defined as a coroutine but `await` is not used anywhere here. Are you sure this is supposed to be a coroutine and not a regular method? I would say that, if any of the commands issued here can take more time to execute than half a heartbeat, then it needs to be a coroutine. If the aforementioned method is not a coroutine, you may need to execute it in a separate thread so you don't compromise the CSC responsiveness. 

Assuming, for instance, that the command `self.model.change_wavelength(id_data.data.wavelength)` takes something between 0 to 60s to execute you have basically two options. 

make `self.model.change_wavelength` a coroutine and replace the line by

```python
await self.model.change_wavelength(id_data.data.wavelength)
```

looking inside the code, I don't think that is possible though. So you are left with the second option, which is to execute it in a separate thread. To do that you can;

```python 
import concurrent

...

executor = concurrent.futures.ThreadPoolExecutor(max_workers=1)
loop = asyncio.get_event_loop()
await loop.run_in_executor(executor, self.model.change_wavelength, id_data.data.wavelength)
```